pupil-spa/src/app/services/user/user.service.ts

```diff
@@ -30,12 +37,14 @@ export class UserService {
         .then(data => {
           this.loggedIn = true;
           this.storageService.clear();
-          this.storageService.setItem(questionsDataKey, data[questionsDataKey]);
-          this.storageService.setItem(configDataKey, data[configDataKey]);
-          this.storageService.setItem(pupilDataKey, data[pupilDataKey]);
-          this.storageService.setItem(schoolDataKey, data[schoolDataKey]);
-          this.storageService.setItem(accessTokenKey, data[tokenKey] && data[tokenKey]['jwt'] && data[tokenKey]['jwt']['token']);
-          this.storageService.setItem(tokenKey, data[tokenKey]);
+          this.storageService.setItem(questionsStorageKey, data[questionsStorageKey.toString()]);
```

Comment:
This usage makes it seem possible to re-use the `questionsDataKey`, which in this case would a very bad idea as these objects can only be used once.  It seems to be asking too much to ask the programmer to remember which objects have a random suffix and which do not.  This seems to be a design limitation that will eventually lead to programmer error.

Reply:
this has been modified now.